### PR TITLE
Support empty `TensorFrame({}, {})`

### PR DESCRIPTION
**Pull request by akihironitta**
This PR adds support for empty `TensorFrame` similarly to `pandas.DataFrame`.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -18,3 +18,4 @@ venv/*
 *.out
 data/**
 catboost_info/
+.mypy_cache
```
Comment (yiweny) [inline .gitignore:21]: Remove?

File: torch_frame/data/tensor_frame.py [hunk 2]
```diff
@@ -84,9 +84,6 @@ def __init__(
 
     def validate(self) -> None:
         r"""Validates the :class:`TensorFrame` object."""
-        if len(self.feat_dict) == 0:
```
Comment (yiweny) [inline torch_frame/data/tensor_frame.py:87]: Can we check instead if not only `feat_dict` or `col_names_dict` is empty?

File: .gitignore (hunk 1 above)
Comment (akihironitta) [reply .gitignore:21]: Decoupled this to another PR #341. PTAL

Comment (akihironitta) [inline .gitignore:21]:
```suggestion
```

File: torch_frame/data/tensor_frame.py (hunk 2 above)
Comment (akihironitta) [reply torch_frame/data/tensor_frame.py:87]: It should already be covered by the check `self.feat_dict.keys() != self.col_names_dict.keys()` right below this line.

**Review by rusty1s — APPROVED** [review]

**Pull request merged**